api/envoy/type/matcher/regex.proto

```diff
@@ -19,6 +19,11 @@ message RegexMatcher {
   // Google's `RE2 <https://github.com/google/re2>`_ regex engine. The regex string must adhere to
   // the documented `syntax <https://github.com/google/re2/wiki/Syntax>`_. The engine is designed
   // to complete execution in linear time as well as limit the amount of memory used.
+  // Envoy supports program size checking via runtime. The runtime keys `regex.max_program_size_error_level`
+  // and `regex.max_program_size_warn_level` can be set to integers as the maximum program size or
```

Comment:
Nit: I would maybe call this `re2.max_program_size.error_level` and `re2.max_program_size.warn_level`., since this a RE2 specific configuration.